src/core/alt-table.rkt

```diff
@@ -145,20 +179,20 @@
 
 (define (atab-prune atab)
   (define sc (atab->set-cover atab))
-  (let loop ([removed '()])
+  (define removable (sort (set->list (set-cover-removable sc)) expr<?
+                          #:key (compose program-body alt-program)))
+  (let loop ([removed '()] [removable removable])
     (if (set-empty? (set-cover-removable sc))
         (apply atab-remove* atab removed)
-        (let ([worst-alt (worst atab (set->list (set-cover-removable sc)))])
+        (let ([worst-alt (worst atab removable)])
           (set-cover-remove! sc worst-alt)
-          (loop (cons worst-alt removed))))))
+          (loop (cons worst-alt removed)
+                (filter (curry set-member? (set-cover-removable sc)) removable))))))
```

Comment:
I'm willing to merge this though it's ultimately pretty ugly.

Reply:
It's a little gross. I suppose I could reconstruct the list each time and not filter it.